### PR TITLE
[scheduler] Simplify time-grid event duration checks

### PR DESCRIPTION
**Pull request by flaviendelangle**
The time-grid event hook split durations into under-30-minute and 30-to-60-minute flags, but both consumers always combined them to decide whether an event was under an hour. Replace the two flags with one under-an-hour check and expose the derived `isStacked` value to regular events and placeholders.

The inclusive 15-minute padding threshold is unchanged.

Validation: 57 tests passed, including the existing 15-, 30-, and 60-minute boundary cases; scheduler premium TypeScript, targeted ESLint, and Prettier passed. `pnpm proptypes` and `pnpm docs:api` completed without generated changes.

- [x] I have followed the PR section of the contributing guide.

### CODE REVIEW

File: packages/x-scheduler/src/internals/components/event/time-grid-event/useTimeGridEvent.ts [hunk 1]
```diff
@@ -101,8 +98,7 @@ export function useTimeGridEvent(
     isEndResizable,
     palette,
     isLessThan15Minutes,
-    isLessThan30Minutes,
-    isBetween30and60Minutes,
+    isStacked: !isUnderHour,
```
Comment (rita-codes) [inline packages/x-scheduler/src/internals/components/event/time-grid-event/useTimeGridEvent.ts:97]: ℹ️ **The hook still returns fields nothing consumes**

Neither `TimeGridEventRegular` nor `TimeGridEventPlaceholder` destructures `palette` or `isLessThan15Minutes`; both already reach the DOM through `rootDataAttributes`. This is pre-existing, but since this PR trims exactly this return shape, it is the natural moment to drop the two dead fields and their entries in `UseTimeGridEventReturnValue`, keeping them as locals feeding `rootDataAttributes`.

**Review by rita-codes — COMMENTED** [review]
# PR review

Nothing merge-blocking. The rewrite is boolean-exact: `d < 30 || (d >= 30 && d < 60)` collapses to `d < 60` for every finite value, and `isStacked` is its negation, so `data-under-hour`, `data-stacked` and the inline/block time layout are unchanged. The hook is internal (only the two variants in `TimeGridEvent.tsx` consume it), the `TimeGridEventContent` memo still receives a primitive boolean, and the `rootDataAttributes` dependency list matches what the memo reads. Verified on the PR head: `@mui/x-scheduler` typecheck, ESLint on the folder, and the jsdom suites for `DayView` (including the 15/30/60-minute threshold tests) and the event components all pass. The two findings below are naming residue the PR leaves behind, not behavior.

## Tests (1)

### 1. 🟡 Threshold test titles still describe the removed 30-minute tiers

**Location:** `packages/x-scheduler/src/day-view/DayView.test.tsx:74`

```tsx
it('marks a 30-minute event as under an hour but not under fifteen minutes (< 30 boundary)', () => {
...
it('does not mark a 60-minute event as under an hour ([30, 60) boundary)', () => {
```

The parentheticals name the `< 30` and `[30, 60)` tiers that only existed because of `isLessThan30Minutes` and `isBetween30and60Minutes`. After this PR there is no 30-minute threshold anywhere in the code, so a reader following the hook comment ("See the `duration thresholds` boundary tests") lands on titles describing boundaries that no longer exist. The assertions themselves are still correct and still cover the only two remaining thresholds (`<= 15` and `< 60`). The three titles also predate the repo convention of starting with "should".

**Fix:** rename the two titles to describe the `< 60` boundary only, and prefix the 15-minute one with "should" while touching the block.

## Simplifications (1)

### 1. ℹ️ The hook still returns fields nothing consumes

See inline comment on `useTimeGridEvent.ts`.

## Verdict

**Approve after nits** — the logic is equivalent and covered, but the threshold test titles now reference tiers the PR deletes.

---

🤖 Review generated with Claude Code

File: packages/x-scheduler/src/internals/components/event/time-grid-event/useTimeGridEvent.ts (hunk 1 above)
Comment (flaviendelangle) [reply packages/x-scheduler/src/internals/components/event/time-grid-event/useTimeGridEvent.ts:97]: Removed `palette` and `isLessThan15Minutes` from the return type and returned object; both remain local inputs to `rootDataAttributes`. Also renamed the duration tests to describe the remaining thresholds and start with "should".

Validated with 57 day-view tests, scheduler TypeScript, targeted ESLint, formatting, PropTypes, and API generation. Fixed in 3ce101f718.

🤖 Comment generated with Codex

**Review by rita-codes — APPROVED** [review]
